java/code/src/com/redhat/rhn/common/db/datasource/CachedStatement.java

```diff
@@ -969,4 +970,14 @@ public DataResult<?> restartQuery() {
             HibernateHelper.cleanupDB(ps);
         }
     }
+
+    /**
+     * Restore the linked hibernate session if it is closed
+     */
+    public void restoreSessionIfClosed() {
+        if (!session.isOpen()) {
+            session = session.getSessionFactory().openSession();
+            session.beginTransaction();
+        }
```

Comment:
```suggestion
        if (session.isOpen()) {
            return;
        }

        session = session.getSessionFactory().openSession();
        session.beginTransaction();
```
Nitpicking here, but you can invert this if and do an early out to avoid both negated logic and code nesting.